File: src/0026-remove-duplicates-from-sorted-array/solution.go
package solution

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	i := 0
	for _, v := range nums {
		if v != nums[i] {
			i++
			nums[i] = v
		}
	}
	nums = nums[:i+1]
	return i + 1
}

File: src/0026-remove-duplicates-from-sorted-array/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    if (nums.length <= 1) {
        return nums.length;
    }
    var num = nums[0];
    for (var i = 1; i < nums.length; i++) {
        if (nums[i] != num) {
            num = nums[i];
            continue;
        }
        nums.splice(i, 1);
        i--;
    }
    return nums.length;
};
